### PR TITLE
Do not mark a switch expression as erroneous just because it wasn't exhaustive.

### LINKED ISSUE #31167: Pattern switch expression crash in emit
The following test fails, crashing in emit.

``` c#
        [Fact]
        public void NonExhaustiveBoolSwitchExpression()
        {
            var source = @"using System;
class Program
{
    static void Main()
    {
        new Program().Start();
    }
    void Start()
    {
        Console.Write(M(true));
        try
        {
            Console.Write(M(false));
        }
        catch (Exception)
        {
            Console.Write("" throw"");
        }
    }
    public int M(bool b) 
    {
        return b switch
        {
           true => 1
        }; 
    }
}
";
            var compilation = CreatePatternCompilation(source);
            compilation.VerifyDiagnostics(
                // (22,18): warning CS8509: The switch expression does not handle all possible inputs (it is not exhaustive).
                //         return b switch
                Diagnostic(ErrorCode.WRN_SwitchExpressionNotExhaustive, "switch").WithLocation(22, 18)
                );
            CompileAndVerify(compilation, expectedOutput: "1 throw");
        }
```

Reported by @jcouv

### PR DESCRIPTION
**Pull request by gafter**
Fixes #31167

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests4.cs [hunk 1]
```diff
@@ -1487,5 +1487,45 @@ static int M1(bool? b1, bool? b2)
                 Diagnostic(ErrorCode.ERR_SwitchArmSubsumed, "(null, true)").WithLocation(13, 13)
                 );
         }
+
+        [Fact]
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests4.cs]: >Fact [](start = 9, length = 4)

link to issue?

File: src/Compilers/CSharp/Portable/Binder/SwitchExpressionBinder.cs [hunk 2]
```diff
@@ -53,7 +53,6 @@ internal override BoundExpression BindSwitchExpressionCore(SwitchExpressionSynta
         /// <param name="switchArms"></param>
         /// <param name="decisionDag"></param>
         /// <param name="diagnostics"></param>
-        /// <returns>true if the switch expression is not exhaustive</returns>
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Binder/SwitchExpressionBinder.cs]: Why remove the comment?

**Comment by gafter** [comment]
@agocke Could you review this please?

**Comment by jcouv** [comment]
@gafter I think you don't need a second sign-off for a test- and comment-only change.

**Pull request merged**